### PR TITLE
parser, session: fix wrong comment in selection field

### LINKED ISSUE #3739: The wrong comment of `select 1 /*!32301 +1 */;`
Please answer these questions before submitting your issue. Thanks!

1. What did you do?
If possible, provide a recipe for reproducing the error.
```
select 1 /*!32301 +1 */;
```

2. What did you expect to see?
```
+-------+
| 1  +1 |
+-------+
|     2 |
+-------+
1 row in set (0.01 sec)
```

3. What did you see instead?
```
+------------------+
| 1 /*!32301 +1 */ |
+------------------+
|                2 |
+------------------+
1 row in set (0.01 sec)
```

4. What version of TiDB are you using (`tidb-server -V`)?

### PR DESCRIPTION
**Pull request by jackysp**
Fix #3739, the wrong comment in selection field.
Due to #2017, I fix the result set fields instead of fix it in parser.
PTAL @tiancaiamao @zimulala

### CODE REVIEW

**Comment by shenli** [comment]
LGTM
@tiancaiamao PTAL

**Comment by tiancaiamao** [comment]
If you don't fix it in parser, maybe you can change here https://github.com/pingcap/tidb/blob/master/plan/logical_plan_builder.go#L427 ?

File: parser/yy_parser.go [hunk 1]
```diff
@@ -38,12 +38,15 @@ const (
...
-	specCodeEnd     = regexp.MustCompile(`[ \t]*\*\/$`)
+	// SpecFieldPattern special result field pattern
+	SpecFieldPattern = regexp.MustCompile(`(\/\*!(M?[0-9]{5,6})?|\*\/)`)
+	specCodePattern  = regexp.MustCompile(`\/\*!(M?[0-9]{5,6})?([^*]|\*+[^*/])*\*+\/`)
+	specCodeStart    = regexp.MustCompile(`^\/\*!(M?[0-9]{5,6})?[ \t]*`)
+	specCodeEnd      = regexp.MustCompile(`[ \t]*\*\/$`)
 )
 
-func trimComment(txt string) string {
+// TrimComment trim comment for special comment code of MySQL
```
Comment (zimulala) [inline parser/yy_parser.go]: s/trim/trims
Add `.` at the end of a line.

**Comment by jackysp** [comment]
@tiancaiamao it's another choice, but it goes a little too deep in logical. I'm afraid it will make it difficult to refactor the logical plan in future.

**Comment by tiancaiamao** [comment]
Just replace `fleid.Text()` with `parser.SpecFieldPattern.ReplaceAllStringFunc(field.Text(), parser.TrimComment)`

If it's not fixed in parser, then the suboptimal place is here.

**Comment by jackysp** [comment]
PTAL @tiancaiamao @zimulala

**Comment by tiancaiamao** [comment]
LGTM

**Pull request merged**